### PR TITLE
CLI: Typescript strict mode

### PR DESCRIPTION
**Pull request by 0916dhkim**
Part of #22176

## What I did
Fix some typescript errors before enabling strict mode.

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [x] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/lib/cli/src/automigrate/fixes/builder-vite.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ export const builderVite: Fix<BuilderViteOptions> = {
 
     logger.info(`✅ Updating main.js to use vite builder`);
     if (!dryRun) {
-      await updateMainConfig({ dryRun, mainConfigPath }, async (main) => {
```
Comment (0916dhkim) [inline code/lib/cli/src/automigrate/fixes/builder-vite.ts:79]: `mainConfigPath` is declared as optional in `RunOptions` here https://github.com/storybookjs/storybook/blob/e7817e82757618992f930f6157ddb68fc7541260/code/lib/cli/src/automigrate/types.ts#L13

but `updateMainConfig` expects non-null string. What would be a good fix here?

File: code/lib/cli/src/automigrate/fixes/eslint-plugin.ts [hunk 2]
```diff
@@ -34,15 +34,15 @@ export const eslintPlugin: Fix<EslintPluginRunOptions> = {
...
+    let unsupportedExtension: string | undefined;
     try {
       eslintFile = findEslintFile();
     } catch (err) {
-      unsupportedExtension = err.message;
+      unsupportedExtension = String(err);
     }
 
-    if (!eslintFile && !unsupportedExtension) {
+    if (!eslintFile || !unsupportedExtension) {
```
Comment (0916dhkim) [inline code/lib/cli/src/automigrate/fixes/eslint-plugin.ts:45]: I'm pretty sure this is meant to be `||` here. Let me know if this is a mistake.

File: code/lib/cli/src/add.ts [hunk 3]
```diff
@@ -56,7 +56,10 @@ const postinstallAddon = async (addonName: string, isOfficialAddon: boolean) =>
 
 const getVersionSpecifier = (addon: string) => {
   const groups = /^(...*)@(.*)$/.exec(addon);
-  return groups ? [groups[1], groups[2]] : [addon, undefined];
+  if (groups) {
+    return [groups[0], groups[2]] as const;
+  }
+  return [addon, undefined] as const;
```
Comment (ryami333) [inline code/lib/cli/src/add.ts:62]: Probably would have just made more sense to add a tuple return-type to the `getVersionSpecifier` function definition.

**Comment by ndelangen** [comment]
@SocketSecurity ignore json-parse-even-better-errors@3.0.0
@SocketSecurity ignore cacache@17.0.6
@SocketSecurity ignore npm-install-checks@6.1.1
@SocketSecurity ignore semver@7.4.0
@SocketSecurity ignore fs-minipass@3.0.2
@SocketSecurity ignore json-stable-stringify@1.0.2
@SocketSecurity ignore url@0.11.0
@SocketSecurity ignore webpack-hot-middleware@2.25.3
@SocketSecurity ignore clone-response@1.0.3
@SocketSecurity ignore responselike@2.0.1
@SocketSecurity ignore danger@11.2.6
@SocketSecurity ignore node-cleanup@2.1.2

**Review by ndelangen — APPROVED** [review]

**Pull request merged**